airflow/dag_processing/processor.py

```diff
@@ -618,6 +619,8 @@ def process_file(
             Stats.incr('dag_file_refresh_error', 1, 1)
             return 0, 0
 
+        self._deactivate_missing_dags(session, dagbag, file_path)
```

Comment:
We should move this line at the end of function (or at the very least after len(dagbad.dags) check so we don't call DB if there are no dags.

Reply:
I think that we _do_ want to call the DB if there's no DAGs in a file - if a file used to have one DAG and now has zero then we'll need to mark that DAG is inactive, correct?

Reply:
hmm yes True